code/modules/mod/mod_modules.dm

```diff
@@ -0,0 +1,616 @@
+	. = ..()
+	if(!.)
+		return
+	ion_trail.stop()
+	UnregisterSignal(mod.wearer, COMSIG_MOVABLE_MOVED)
+	UnregisterSignal(mod.wearer, COMSIG_MOVABLE_PRE_MOVE)
+	UnregisterSignal(mod.wearer, COMSIG_MOVABLE_SPACEMOVE)
+	mod.wearer.remove_movespeed_modifier(/datum/movespeed_modifier/jetpack/fullspeed)
+
+/obj/item/mod/module/jetpack/proc/move_react(mob/user)
+	if(!active)//If jet dont work, it dont work
+		return
+	if(!mod.wearer?.client)//Don't allow jet self using
+		return
+	if(!isturf(mod.wearer.loc))//You can't use jet in nowhere or from mecha/closet
+		return
+	if(!(mod.wearer.movement_type & FLOATING) || mod.wearer.buckled)//You don't want use jet in gravity or while buckled.
+		return
+	if(mod.wearer.pulledby)//You don't must use jet if someone pull you
+		return
+	if(mod.wearer.throwing)//You don't must use jet if you thrown
+		return
+	if(length(mod.wearer.client.keys_held & mod.wearer.client.movement_keys))//You use jet when press keys. yes.
+		allow_thrust()
+
+/obj/item/mod/module/jetpack/proc/pre_move_react(mob/user)
+	ion_trail.oldposition = get_turf(src)
+
+/obj/item/mod/module/jetpack/proc/spacemove_react(mob/user, movement_dir)
+	SIGNAL_HANDLER
+
+	if(active && movement_dir)
+		return COMSIG_MOVABLE_STOP_SPACEMOVE
+
+/obj/item/mod/module/jetpack/proc/allow_thrust()
+	if(!drain_power(use_power_cost))
+		return
+	ion_trail.generate_effect()
+	return TRUE
+
+/obj/item/mod/module/magboot
+	name = "MOD magnetic stability module"
+	desc = "A module granting magnetic stability to the wearer, protecting them from forces pushing them away."
+	module_type = MODULE_TOGGLE
+	complexity = 2
+	active_power_cost = 20
+	incompatible_modules = list(/obj/item/mod/module/magboot)
+	cooldown_time = 0.5 SECONDS
+	var/slowdown_active = 1
+
+/obj/item/mod/module/magboot/on_activation()
+	. = ..()
+	if(!.)
+		return
+	ADD_TRAIT(mod.wearer, TRAIT_NEGATES_GRAVITY, MOD_TRAIT)
+	mod.slowdown += slowdown_active
+	mod.wearer.update_equipment_speed_mods()
+	mod.wearer.update_gravity(mod.wearer.has_gravity())
+
+/obj/item/mod/module/magboot/on_deactivation()
+	. = ..()
+	if(!.)
+		return
+	REMOVE_TRAIT(mod.wearer, TRAIT_NEGATES_GRAVITY, MOD_TRAIT)
+	mod.slowdown -= slowdown_active
+	mod.wearer.update_equipment_speed_mods()
+	mod.wearer.update_gravity(mod.wearer.has_gravity())
+
+/obj/item/mod/module/holster
+	name = "MOD holster module"
+	desc = "A module that can instantly holster a gun inside the MOD."
+	module_type = MODULE_USABLE
+	complexity = 2
+	use_power_cost = 25
+	incompatible_modules = list(/obj/item/mod/module/holster)
+	cooldown_time = 0.5 SECONDS
+	var/obj/item/gun/holstered
+
+/obj/item/mod/module/holster/on_use()
+	. = ..()
+	if(!.)
+		return
+	if(!holstered)
+		var/obj/item/gun/holding = mod.wearer.get_active_held_item()
+		if(!holding)
+			to_chat(mod.wearer, "<span class='notice'>You aren't holding anything to holster!</span>")
+			return
+		if(!istype(holding) || holding.w_class > WEIGHT_CLASS_BULKY || holding.weapon_weight > WEAPON_MEDIUM)
```

Comment:
are these the same restrictions that the exosuit storage slot uses for guns?

Reply:
speaking of the exosuit storage slot, why not just use that instead of making the gun disappear and reappear? then people would be able to hotkey it out and stuff, and you could see a dangerous gun when examining someones's modsuit (I think)

Reply:
because this is cooler and i like it